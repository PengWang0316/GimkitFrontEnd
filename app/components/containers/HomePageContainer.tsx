import React, { memo, useCallback, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Fab, Tooltip } from '@material-ui/core';
import { Add as AddIcon } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import PostList from '../PostList';
import AddPostDialog from '../AddPostDialog';

const useStyles = makeStyles({
  rootPaper: {
    margin: '50px auto 10px auto',
    width: '80%',
    padding: 15,
  },
  fab: {
    position: 'fixed',
    bottom: 20,
    right: 20,
  },
});

export const HomePageContainer = () => {
  const classes = useStyles({});
  const [isOpenAddPost, setIsOpenAddPost] = useState(false);
  const addPostDialogCloseCallback = useCallback(() => setIsOpenAddPost((state: boolean) => !state), []);

  return (
    <Paper className={classes.rootPaper}>
      <PostList />
      <Tooltip title={I18n.get('addPost')} placement="left-start">
        <Fab size="medium" color="secondary" aria-label="Add" className={classes.fab} onClick={addPostDialogCloseCallback}>
          <AddIcon />
        </Fab>
      </Tooltip>
      <AddPostDialog
        isOpen={isOpenAddPost}
        handleClose={addPostDialogCloseCallback}
      />
    </Paper>
  );
};

export default memo(HomePageContainer);
