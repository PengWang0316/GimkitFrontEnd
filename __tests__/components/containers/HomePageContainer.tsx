import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { HomePageContainer } from '../../../app/components/containers/HomePageContainer';

jest.mock('@material-ui/core/Paper', () => 'Paper');
jest.mock('@material-ui/core/Fab', () => 'Fab');
jest.mock('@material-ui/core/Tooltip', () => 'Tooltip');
jest.mock('@material-ui/icons/Add', () => 'Add');
jest.mock('@kevinwang0316/i18n', () => ({ get: jest.fn().mockReturnValue('I18n text') }));
jest.mock('../../../app/components/PostList', () => 'PostList');
jest.mock('../../../app/components/AddPostDialog', () => 'AddPostDialog');

describe('HomePageContainer', () => {
  const defaultProps = {};

  const getShallowComponent = (props = defaultProps) => shallow(<HomePageContainer {...props} />);
  test('snapshot', () => expect(renderer.create(<HomePageContainer />).toJSON()).toMatchSnapshot());
});
