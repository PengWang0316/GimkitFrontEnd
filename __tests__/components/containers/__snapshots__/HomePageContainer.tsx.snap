// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContainer snapshot 1`] = `
<Paper
  className="makeStyles-rootPaper-1"
>
  <PostList />
  <Tooltip
    placement="left-start"
    title="I18n text"
  >
    <Fab
      aria-label="Add"
      className="makeStyles-fab-2"
      color="secondary"
      onClick={[Function]}
      size="medium"
    >
      <Add />
    </Fab>
  </Tooltip>
  <AddPostDialog
    handleClose={[Function]}
    isOpen={false}
  />
</Paper>
`;
